eng/targets/Imports.targets

```diff
@@ -43,9 +43,10 @@
     <!--
       Disable code style analyzers in "older" targets for a multi-targeted project. These analyzers don't
       impact the correctness of the output, so we avoid the performance overhead where it's easy to do so.
+      Enable code style analyzers for design time builds for live background analysis while typing (see https://github.com/dotnet/project-system/blob/main/docs/design-time-builds.md#determining-whether-a-target-is-running-in-a-design-time-build).
     -->
     <RoslynCheckCodeStyle Condition="'$(RoslynCheckCodeStyle)' == '' AND '$(DisableNullableWarnings)' == 'true'">false</RoslynCheckCodeStyle>
-    <RoslynCheckCodeStyle Condition="'$(RoslynCheckCodeStyle)' == '' AND ('$(ContinuousIntegrationBuild)' != 'true' OR '$(RoslynEnforceCodeStyle)' == 'true')">true</RoslynCheckCodeStyle>
+    <RoslynCheckCodeStyle Condition="'$(RoslynCheckCodeStyle)' == '' AND ('$(ContinuousIntegrationBuild)' != 'true' OR '$(RoslynEnforceCodeStyle)' == 'true' OR '$(DesignTimeBuild)' == 'true' OR '$(BuildingProject)' != 'true')">true</RoslynCheckCodeStyle>
```

Comment:
This change is not necessary because `ContinuousIntegrationBuild` will never be true for design time builds.

Reply:
@sharwell Without this change, this check seems to fail and we are using the IDExxxx analyzers/fixers from Features.dll

Reply:
You can quickly validate this by looking at the Analyzers node for net6/net7 targets and confirm the CodeStyle analyzer assemblies do not show up under this node without this change, and does so after this change.

Reply:
Did you somehow locally set ContinuousIntegrationBuild to true (e.g. via environment variable)?

Reply:
Alright, I did a git clean after reverting the change in this file and now do see the CodeStyle analyzer reference. I'll revert this part of the PR.